test/testobjects/testhistories.ts

```diff
@@ -1313,127 +1293,148 @@ export class TestHistories {
+                timestamp,
                 isPlayed: false,
             }),
             new TimerFiredEvent({
                 eventId: -1,
-                timestamp: firstMoment.add(2, "s").add(retryInterval, "ms").toDate(),
+                timestamp,
                 isPlayed: false,
-                fireAt: firstMoment.add(2, "s").add(retryInterval, "ms").toDate(),
+                fireAt: timestamp,
                 timerId: 1,
             }),
             new TaskScheduledEvent({
                 eventId: 2,
-                timestamp: firstTimestamp,
+                timestamp,
                 isPlayed: false,
                 name: "Hello",
                 input: undefined,
             }),
             new OrchestratorCompletedEvent({
                 eventId: -1,
-                timestamp: firstTimestamp,
+                timestamp,
                 isPlayed: false,
-            }),
+            })
+        );
+
+        timestamp = moment(timestamp).add(20, "s").toDate();
+
+        history.push(
             new OrchestratorStartedEvent({
                 eventId: -1,
-                timestamp: firstMoment.add(3, "s").toDate(),
+                timestamp,
                 isPlayed: false,
             }),
             new TaskFailedEvent({
                 eventId: -1,
-                timestamp: firstMoment.add(3, "s").toDate(),
+                timestamp,
                 isPlayed: false,
                 taskScheduledId: 2,
                 details: "Big stack trace here",
                 reason: "Activity function 'Hello' failed: Result: Failure",
             }),
             new TimerCreatedEvent({
                 eventId: 3,
-                timestamp: firstMoment.add(3, "s").toDate(),
+                timestamp,
                 isPlayed: false,
-                fireAt: firstMoment.add(3, "s").add(retryInterval, "ms").toDate(),
+                fireAt: moment(timestamp).add(retryInterval, "ms").toDate(),
             }),
             new OrchestratorCompletedEvent({
                 eventId: -1,
-                timestamp: firstMoment.add(3, "s").toDate(),
+                timestamp,
                 isPlayed: false,
-            }),
+            })
+        );
+
+        timestamp = moment(timestamp).add(retryInterval, "ms").toDate();
+
+        history.push(
             new OrchestratorStartedEvent({
                 eventId: -1,
-                timestamp: firstMoment.add(4, "s").add(retryInterval, "ms").toDate(),
+                timestamp,
                 isPlayed: false,
             }),
             new TimerFiredEvent({
                 eventId: -1,
-                timestamp: firstMoment.add(4, "s").add(retryInterval, "ms").toDate(),
+                timestamp,
                 isPlayed: false,
-                fireAt: firstMoment.add(4, "s").add(retryInterval, "ms").toDate(),
+                fireAt: timestamp,
                 timerId: 3,

```

Comment:
nitpick: It seems that the timers are firing _exactly_ at their `fireAt` value (value for `timestamp` is the same as `fireAt`), which doesn't usually match reality. In reality, the timestamp of a timer might be a few seconds after it was scheduled to fire.

Just for good practice, can you add some artificial delay in the `timestamp` value of timer fired events? I want our tests to catch errors when timers are "delayed" in their firing for whatever reason.

Reply:
Done :)